henson-plugin/src/main/groovy/dart/henson/plugin/HensonPlugin.groovy

```diff
@@ -39,10 +40,10 @@ class HensonPlugin implements Plugin<Project> {
             File destinationFolder =
                     project.file(
                             new File(project.getBuildDir(), "generated/source/navigator/" + variant.getName()))
-            GenerateHensonNavigatorTask navigatorTask = hensonManager
+            TaskProvider<GenerateHensonNavigatorTask> navigatorTask = hensonManager
                     .createHensonNavigatorGenerationTask(variant, destinationFolder)
 
-            variant.registerJavaGeneratingTask(navigatorTask, destinationFolder)
+            variant.registerJavaGeneratingTask(navigatorTask.get(), destinationFolder)
```

Comment:
This is a bit sad because registerJavaGeneratingTask doesn't accept a task provider. I created a ticket with the android team to support this in the future.